functions/issue-scraper.ts

```
@@ -0,0 +1,351 @@
+  supabase: SupabaseClient,
+  issues: Array<{
+    id: string;
+    markdown: string;
+    plaintext: string;
+    embedding: string;
+    author_id: number;
+    payload: PayloadType;
+  }>
+): Promise<void> {
+  const { error } = await supabase.from("issues").upsert(issues);
+  if (error) {
+    throw new Error(`Error during batch upsert: ${error.message}`);
+  }
+}
+
+async function batchFetchAuthorIds(octokit: InstanceType<typeof Octokit>, logins: string[]): Promise<Record<string, number>> {
+  const authorIdMap: Record<string, number> = {};
+  const BATCH_SIZE = 20;
+  for (let i = 0; i < logins.length; i += BATCH_SIZE) {
+    const batch = logins.slice(i, i + BATCH_SIZE);
+    const promises = batch.map(async (login) => {
+      try {
+        const response = await octokit.rest.users.getByUsername({ username: login });
+        return { login, id: response.data.id };
+      } catch (error) {
+        console.error(`Error fetching author ID for ${login}:`, error);
+        return { login, id: -1 };
+      }
+    });
+    const results = await Promise.all(promises);
+    results.forEach(({ login, id }) => {
+      authorIdMap[login] = id;
+    });
+  }
+  return authorIdMap;
+}
+
+async function issueScraper(username: string, supabase: SupabaseClient, voyageApiKey: string, token?: string, timestamp?: number): Promise<string> {
+  try {
+    if (!username) {
+      throw new Error("Username is required");
+    }
+
+    const octokit = new Octokit(token ? { auth: token } : {});
+    const voyageClient = new VoyageAIClient({ apiKey: voyageApiKey });
+
+    const issues = await fetchUserIssuesBatch(octokit, username, timestamp);
+
+    const uniqueAuthors = Array.from(new Set(issues.map((issue) => issue.author?.login).filter((login): login is string => !!login)));
+
+    const authorIdMap = await batchFetchAuthorIds(octokit, uniqueAuthors);
+
+    const markdowns = issues.map((issue) => `${issue.body || ""} ${issue.title || ""}`);
```

I think that titles can NOT be empty so this should throw an error.